Core/HLE/sceKernelThread.cpp

```diff
@@ -3095,6 +3100,13 @@ void __KernelExecuteMipsCallOnCurrentThread(u32 callId, bool reschedAfter)
 	call->savedV1 = currentMIPS->r[MIPS_REG_V1];
 	call->savedId = cur->currentMipscallId;
 	call->reschedAfter = reschedAfter;
+	
+	// Also need to backup other regs which might be used by the called mips function, needed to prevent corrupting the original regs (ie. Dissidia 012 with more than 2 players, Lord of Arcana)
```

Comment:
The thing is AfterMatchingMipsCall didn't always get executed, even within a few minutes timeout when i tested it (i used 250ms as matching callback timeout as it could cause slowdown/lag if i set too high, since most of the time it only takes less than 100ms), i believed the failing AfterMatchingMipsCall was caused by corrupted registers/memory.